packages/backend/src/dbt/profiles.ts

```diff
@@ -26,31 +27,47 @@ const credentialsTarget = (
 ): CredentialsTarget => {
     switch (credentials.type) {
         case WarehouseTypes.BIGQUERY:
-            return {
+            const bqResult: CredentialsTarget = {
                 target: {
                     type: credentials.type,
-                    method: 'service-account-json',
                     project: credentials.project,
                     dataset: credentials.dataset,
                     threads: DEFAULT_THREADS,
                     timeout_seconds: credentials.timeoutSeconds,
                     priority: credentials.priority,
                     retries: credentials.retries,
                     maximum_bytes_billed: credentials.maximumBytesBilled,
-                    keyfile_json: Object.fromEntries(
-                        Object.keys(credentials.keyfileContents).map((key) => [
-                            key,
-                            envVarReference(key),
-                        ]),
-                    ),
                     execution_project: credentials.executionProject,
                 },
-                environment: Object.fromEntries(
+                environment: {},
+            };
+            if (credentials.keyfileContents) {
+                bqResult.target.method = 'service-account-json';
+                bqResult.target.keyfile_json = Object.fromEntries(
+                    Object.keys(credentials.keyfileContents).map((key) => [
+                        key,
+                        envVarReference(key),
+                    ]),
+                );
+                bqResult.environment = Object.fromEntries(
                     Object.entries(credentials.keyfileContents).map(
                         ([key, value]) => [envVar(key), value],
                     ),
-                ),
-            };
+                );
+                return bqResult;
+            }
+            if (
+                credentials.authenticationType ===
+                BigqueryAuthenticationType.ADC
+            ) {
+                // With oauth method and no keyfile contents, dbt will use the
+                // application default credentials (ADC) to authenticate
+                bqResult.target.method = 'oauth';
+                return bqResult;
+            }
+            throw new Error(
```

Comment:
Normally we do `switch` and `assertUnreachable` on default, this will also helps us during compile, rather than throwing runtime errors.

Reply:
@rephus Thank you for your comment!

I noticed that we need to handle `undefined` case, which may happen when we 'test & deploy' old BigQuery project without any update. For now, I've decided to treat this case the same way as private key. Please let me know if you have a different opinion.